pkg/startup/startup.go

```diff
@@ -630,6 +632,15 @@ func autoDetectCIDR(method string, version int) *cnet.IPNet {
 		// Regexes are passed in as a string separated by ","
 		ifRegexes := regexp.MustCompile(`\s*,\s*`).Split(ifStr, -1)
 		return autoDetectCIDRByInterface(ifRegexes, version)
+	} else if strings.HasPrefix(method, AUTODETECTION_METHOD_CIDR) {
+		// Autodetect the IP by filtering interface by its address.
+		cidrStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_CIDR)
+		// CIDRs are passed in as a string separated by ","
+		matches:=[]cnet.IPNet{}
+		for _, r:= range regexp.MustCompile(`\s*,\s*`).Split(cidrStr, -1) {
+			matches=append(matches, cnet.MustParseCIDR(r))
```

Comment:
We shouldn't use `MustParseCIDR` here - that will cause a panic if the input is not a valid CIDR.

Instead, we should parse and handle the error gracefully.

Reply:
fixed